lib/block-supports/behaviors.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Behaviors block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers the behaviors block attribute for block types that support it.
+ * And add the needed hooks to add those behaviors.
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function gutenberg_register_behaviors_support( $block_type ) {
+	$has_behaviors_support = block_has_support( $block_type, array( 'behaviors' ), false );
+	if ( ! $has_behaviors_support ) {
+		return;
+	}
+
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	$block_type->attributes['behaviors'] = array(
+		'type' => 'object',
+	);
+
+	// If it supports the lightbox behavior, add the hook to that block.
+	// In the future, this should be a loop with all the behaviors.
+	$has_lightbox_support = block_has_support( $block_type, array( 'behaviors', 'lightbox' ), false );
+	if ( $has_lightbox_support ) {
+		add_filter( 'render_block_' . $block_type->name, 'gutenberg_blocks_supports_add_lightbox', 15, 2 );
+	}
+}
+
+/**
+ * Add the directives and layout needed for the lightbox behavior.
+ * This functions shouldn't be in this file. It should be moved to a package (or somewhere else), where all the behaviors logic is defined.
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function gutenberg_blocks_supports_add_lightbox( $block_content, $block ) {
```

Comment:
Small nit. All of the other block supports that update rendering follow the same naming pattern which makes it a little easier to search through.

`gutenberg_render_behaviors_support` would match the pattern here.

Reply:
Oh, great! I will make that change as well 🙂